### ISSUE TITLE
[no-restricted-globals] False positive with optional chaining

### ISSUE THREAD

**Issue by m-rutter** [typescript-eslint/typescript-eslint#1090]
I wasn't quite sure which should sub-project was appropriate for this issue. I am aware that strictly ts 3.7.0-beta isn't supported while TS 3.7 syntax is, but out of caution I thought to raise the issue as I couldn't see any other reports.

**What code were you trying to parse?**

```ts
// foobar.ts
const a = foo?.bar?.name ?? "foobar"
```

**What did you expect to happen?**

That eslint would not emit any warnings nor errors

**What actually happened?**

Eslint emitted the following error: 
`Unexpected use of 'name'  no-restricted-globals`

This does not happen for all properties, but so far I've only been able to reproduce with properties called `name`.

**Versions**

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/parser` | `2.4.0` |
| `TypeScript`                           | `3.7.0-beta` (15 Oct 2019) |
| `ESLint`                    | `6.5.1` |
| `node`                                 | `12.11.1` |
| `npm`                                  | `6.11.3` |
| `yarn`                                  | `1.19.1` |

**Comment by bradzacher** [comment]
Thinking about it - this is happening for the same reason that #1104 is happening.

They both rely upon scope analysis to determine variables in the program.
With the AST change required to support optional chaining, this caused the scope analysis no longer consider the `name` `Identifier` to be a member, so it instead considers it a variable.

**Comment by aecorredor** [comment]
Same thing happens to me in `3.7.2`. Just with properties called `name`.

**Comment by aecorredor** [comment]
Still happens to me on `2.19.0`. Only when referencing a property called `name`.

**Comment by bradzacher** [comment]
That would be because `name` is a global in the browser environment, hence it is marked as restricted by the rule.
https://eslint.org/docs/user-guide/configuring#specifying-environments

**Comment by aecorredor** [comment]
@bradleyayer mmm ok I get that. The weird thins is that it's happening to me when I reference it from a JSON object that has some data I imported.

Example:

```js
import banks from './data/banks.json'

const getBankNameByCode = code => {
   const selectedBank = banks.find(bank => bank.code === code);
   return selectedBank?.name ?? ''; // error happens here.
}
```

**Comment by bradzacher** [comment]
I am unable to repro this against master.

![image](https://user-images.githubusercontent.com/7462525/73768883-a32e2700-472e-11ea-9f61-c78849a2cf99.png)

https://github.com/typescript-eslint/typescript-eslint/blob/10d86b1f648896c418041e5eaee5198d274e7d27/packages/eslint-plugin/tests/eslint-rules/no-restricted-globals.test.ts#L33-L42

**Comment by aecorredor** [comment]
@bradzacher thanks for checking. I'll dig in more on my side and report back here if anything. Maybe it's related to my specific project config for some reason.
